set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java

```diff
@@ -217,55 +207,44 @@
 	@RequestMapping(value = {"/set/{identifier}"}, method = RequestMethod.PUT, 
 			produces = {HttpHeaders.CONTENT_TYPE_JSONLD_UTF8, HttpHeaders.CONTENT_TYPE_JSON_UTF8})
 	@ApiOperation(notes = SwaggerConstants.UPDATE_SAMPLES_JSONLD, value = "Update an existing user set", nickname = "update", response = java.lang.Void.class)
-	public ResponseEntity<String> updateUserSet(@RequestParam(value = WebUserSetFields.PARAM_WSKEY, required = false) String wskey,
+	public ResponseEntity<String> updateUserSet(
 			@PathVariable(value = WebUserSetFields.PATH_PARAM_SET_ID) String identifier,
 			@RequestBody String userSet,
-			@RequestParam(value = WebUserSetFields.USER_TOKEN, required = false, defaultValue = WebUserSetFields.USER_ANONYMOUNS) String userToken,
 			@RequestParam(value = WebUserSetFields.PROFILE, required = false, defaultValue = WebUserSetFields.PROFILE_MINIMAL) String profile,			
 			HttpServletRequest request
 			) throws HttpException {
 		
 		String action = "put:/set/{identifier}";
-		return updateUserSet(request, wskey, identifier, userSet, userToken, profile, action);
+		return updateUserSet(request, identifier, userSet, profile, action);
 	}
 		
 	/**
 	 * This method validates input values, retrieves user set object and
 	 * updates it.
 	 * 
 	 * @param request
-	 * @param wskey The API key
 	 * @param identifier The identifier
-	 * @param userSet The user set fields to update in JSON format e.g. title or description
 	 * @param profile The profile definition
 	 * @param action The action describing the request
 	 * @return response entity that comprises response body, headers and status code
 	 * @throws HttpException
 	 */
-	protected ResponseEntity<String> updateUserSet(HttpServletRequest request, String wsKey, String identifier,
-			String userSetJsonLdStr, String userToken, String profileStr, String action) throws HttpException {
+	protected ResponseEntity<String> updateUserSet(HttpServletRequest request, String identifier,
+			String userSetJsonLdStr, String profileStr, String action) throws HttpException {
 
 		try {
 			// check user credentials, if invalid respond with HTTP 401,
-			// check client access (a valid "wskey" must be provided)
-			// Check client access (a valid "wskey" must be provided)
-			validateApiKey(wsKey);
+		    // verify access rights
+			verifyWriteAccess(Operations.UPDATE, request);
```

Comment:
this method needs to be invoked in the calling method

Reply:
verifyWriteAccess invoked as suggested